Update TrkQual for python-based trainings

AndrewEdmonds11 opened the pull request:
Hi Everyone,

In this PR, I've made code changes so that we can use python-based trainings for TrkQual.

A few updates from DocDB-47519:
 - I haven't removed the ```TrkQual``` data product since it is being used by a couple of modules. I'm waiting to hear back from their maintainers about what to do. I can put in a separate PR for that work
 - I couldn't find an already existing ```MVAResult```-type class so I've kept it here
 - I haven't removed the ProditionsService stuff that exists for TrkQual. I will think about removing it again when I know what to do about the TrkQual data product

Cheers,
Andy

FNALbuild commented:
Hi @AndrewEdmonds11,
You have proposed changes to files in these packages:
- RecoDataProducts
- TrkDiag

which require these tests: build.

@Mu2e/fnalbuild-users, @Mu2e/write have access to CI actions on [main](/Mu2e/Offline/tree/main).





:hourglass: The following tests have been triggered for a016c0d21d5dc7dd6bb3e4a31f62484f61e8d0ae: build  (Build queue is empty)


[About FNALbuild.](https://mu2ewiki.fnal.gov/wiki/GitHubWorkflow#GitHub_Pull_Request_Procedures_and_FNALbuild) [Code review on Mu2e/Offline.](https://mu2ewiki.fnal.gov/wiki/GitHubWorkflow#Code_Review)

FNALbuild commented:
:sunny: The build tests passed at a016c0d21d5dc7dd6bb3e4a31f62484f61e8d0ae.


| Test          | Result        | Details |
| ------------- |:-------------:| ------- |
| test with | :white_check_mark: | Command did not list any other PRs to include |
| merge | :white_check_mark: | Merged a016c0d21d5dc7dd6bb3e4a31f62484f61e8d0ae at e777d06a09eb2be76473d168099927ebeacd01a2 |
| build (prof) | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1932/artifact/scons.log). Build time: 10 min 44 sec |
| ceSimReco | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1932/artifact/ceSimReco.log)  |
| g4test_03MT | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1932/artifact/g4test_03MT.log)  |
| transportOnly | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1932/artifact/transportOnly.log)  |
| POT | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1932/artifact/POT.log)  |
| g4study | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1932/artifact/g4study.log)  |
| cosmicSimReco | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1932/artifact/cosmicSimReco.log)  |
| cosmicOffSpill | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1932/artifact/cosmicOffSpill.log)  |
| ceSteps | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1932/artifact/ceSteps.log)  |
| ceDigi | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1932/artifact/ceDigi.log)  |
| muDauSteps | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1932/artifact/muDauSteps.log)  |
| ceMix | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1932/artifact/ceMix.log)  |
| rootOverlaps | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1932/artifact/rootOverlaps.log)  |
| g4surfaceCheck | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1932/artifact/g4surfaceCheck.log)  |
| FIXME, TODO | :white_check_mark: | [TODO (0) FIXME (0) in 2 files](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1932/artifact/fixme_todo.log) |
| clang-tidy | :large_orange_diamond: | [0 errors 211 warnings](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1932/artifact/clang-tidy.log) |
| whitespace check | :white_check_mark: | [no whitespace errors found](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1932/artifact/whitespace_errs.log) |


N.B. These results were obtained from a build of this Pull Request at a016c0d21d5dc7dd6bb3e4a31f62484f61e8d0ae after being merged into the base branch at e777d06a09eb2be76473d168099927ebeacd01a2.


For more information, please check the job page [here](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1932/console).
Build artifacts are deleted after 5 days. If this is not desired, select `Keep this build forever` on the job page.

kutschke reviewed (commented):
I think that you also need to edit:
  RecoDataProducts/src/classes.h
  RecoDataProducts/src/classes_def.xml 

Without that you won't be able to write data products that use this class.  Or am I missing something

Any thoughts on a policy to document where the trkqual.dat and .hxx came from?  For example a tag in your repo, plus names of input files used for training?    Where would it go - in the commit comments?  Somewhere else?

AndrewEdmonds11 commented on RecoDataProducts/inc/MVAResult.hh:
```diff
@@ -0,0 +1,17 @@
+//
+// A datsa product that stores the result of any MVA/ML/AI algorithm
```
Fixed in 778a36b76

AndrewEdmonds11 commented:
Thanks, Rob. Answers below:

> I think that you also need to edit: RecoDataProducts/src/classes.h RecoDataProducts/src/classes_def.xml
> 
> Without that you won't be able to write data products that use this class. Or am I missing something
> 

Good catch. We weren't writing out TrkQual to an art file (it runs in the same job as TrkAna) so I didn't catch this in any of my tests. I've updated the classes.h and classes_def.xml (commit 87983abca) in case we want to do this in the future

> Any thoughts on a policy to document where the trkqual.dat and .hxx came from? For example a tag in your repo, plus names of input files used for training? Where would it go - in the commit comments? Somewhere else?

I've added a reference to the TrkQual repo commit to the top of the .hxx file (commit 6db7e33e0). Unfortunately, we can't add a comment to the .dat file, which may be more important (since we can retrain the same model for a different .dat but use the same .hxx). The input dataset is given in the python notebook so that is documented.

Longer term, we probably want a policy for all our ML algorithms to have the python notebook in DocDB with the input variable distributions, training dataset etc. all documented. Perhaps also referencing it on the Mu2eWiki?

kutschke commented:
@FNALbuild run build test

FNALbuild commented:
:hourglass: The following tests have been triggered for 778a36b76c6a3e3f6d4bff7af24918d5ecdede19: build  (Build queue is empty)

kutschke commented:
> > Any thoughts on a policy to document where the trkqual.dat and .hxx came from? For example a tag in your repo, plus names of input files used for training? Where would it go - in the commit comments? Somewhere else?
> 
> I've added a reference to the TrkQual repo commit to the top of the .hxx file (commit [6db7e33](https://github.com/Mu2e/Offline/commit/6db7e33e03cee906c0bb2e732aba61e370d472e7)). Unfortunately, we can't add a comment to the .dat file, which may be more important (since we can retrain the same model for a different .dat but use the same .hxx). The input dataset is given in the python notebook so that is documented.
> 
> Longer term, we probably want a policy for all our ML algorithms to have the python notebook in DocDB with the input variable distributions, training dataset etc. all documented. Perhaps also referencing it on the Mu2eWiki?

I like the idea of docdb  and prefer it to the wiki - it gives you a natural way to document version evolution.  Then put the docdb number, including version number, as a comment in the .hxx file. Docdb and the wiki both give you a place to put as detailed a writeup as is needed.  Can you do that for this PR?

Can you asl the authors about comments in the .dat files?  If it's not supprted can you request it as a new feature?

Unless comments in the .dat come through for us, there is one thing that might not be robust in this system.  How to document which .dat files can be used with which .hxx file.   We can document it in docdb but people have to remember to do it ( I mean if you retrain with new input files but an otherwise identical model, you will produce a new .dat file that can be read with the same .hxx file, right?)

I guess the solution is that review must check for proper documentation.  Do you see a more robust solution?

FNALbuild commented:
:sunny: The build tests passed at 778a36b76c6a3e3f6d4bff7af24918d5ecdede19.


| Test          | Result        | Details |
| ------------- |:-------------:| ------- |
| test with | :white_check_mark: | Command did not list any other PRs to include |
| merge | :white_check_mark: | Merged 778a36b76c6a3e3f6d4bff7af24918d5ecdede19 at 8926a71d94df388856f3bad5129def074131e278 |
| build (prof) | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1934/artifact/scons.log). Build time: 11 min 18 sec |
| ceSimReco | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1934/artifact/ceSimReco.log)  |
| g4test_03MT | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1934/artifact/g4test_03MT.log)  |
| transportOnly | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1934/artifact/transportOnly.log)  |
| POT | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1934/artifact/POT.log)  |
| g4study | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1934/artifact/g4study.log)  |
| cosmicSimReco | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1934/artifact/cosmicSimReco.log)  |
| cosmicOffSpill | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1934/artifact/cosmicOffSpill.log)  |
| ceSteps | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1934/artifact/ceSteps.log)  |
| ceDigi | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1934/artifact/ceDigi.log)  |
| muDauSteps | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1934/artifact/muDauSteps.log)  |
| ceMix | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1934/artifact/ceMix.log)  |
| rootOverlaps | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1934/artifact/rootOverlaps.log)  |
| g4surfaceCheck | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1934/artifact/g4surfaceCheck.log)  |
| FIXME, TODO | :white_check_mark: | [TODO (0) FIXME (0) in 2 files](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1934/artifact/fixme_todo.log) |
| clang-tidy | :large_orange_diamond: | [0 errors 211 warnings](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1934/artifact/clang-tidy.log) |
| whitespace check | :white_check_mark: | [no whitespace errors found](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1934/artifact/whitespace_errs.log) |


N.B. These results were obtained from a build of this Pull Request at 778a36b76c6a3e3f6d4bff7af24918d5ecdede19 after being merged into the base branch at 8926a71d94df388856f3bad5129def074131e278.


For more information, please check the job page [here](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1934/console).
Build artifacts are deleted after 5 days. If this is not desired, select `Keep this build forever` on the job page.

rlcee commented:
> How to document which .dat files can be used with which .hxx file

Name the hxx
TrkQual_typeN.hxx
Name the dat:
TrkQual_typeN_vM.dat

Any typeN can read any typeN dat.  This also allows you to commit multiple version of hxx. typeN could increment N, or be some other mnemonic.

kutschke commented on RecoDataProducts/src/classes_def.xml:
```diff
@@ -245,6 +245,10 @@
+ <class name="mu2e::MVAResult"/>
+ <class name="mu2e::MVAResultCollection"/>
+ <class name="art::Wrapper<mu2e::MVAResultCollection>"/>
+
```
If you want to make a data product that is a single MVAResult, you also need:

```
<class name="art::Wrapper<mu2e::MVAResult>"/>
```

kutschke commented:
@FNALbuild run build test

FNALbuild commented:
:hourglass: The following tests have been triggered for 949ee9a1dabde5114ad1083e35b7c2108ad0c77a: build  (Build queue is empty)

FNALbuild commented:
:sunny: The build tests passed at 949ee9a1dabde5114ad1083e35b7c2108ad0c77a.


| Test          | Result        | Details |
| ------------- |:-------------:| ------- |
| test with | :white_check_mark: | Command did not list any other PRs to include |
| merge | :white_check_mark: | Merged 949ee9a1dabde5114ad1083e35b7c2108ad0c77a at 8926a71d94df388856f3bad5129def074131e278 |
| build (prof) | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1935/artifact/scons.log). Build time: 11 min 04 sec |
| ceSimReco | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1935/artifact/ceSimReco.log)  |
| g4test_03MT | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1935/artifact/g4test_03MT.log)  |
| transportOnly | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1935/artifact/transportOnly.log)  |
| POT | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1935/artifact/POT.log)  |
| g4study | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1935/artifact/g4study.log)  |
| cosmicSimReco | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1935/artifact/cosmicSimReco.log)  |
| cosmicOffSpill | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1935/artifact/cosmicOffSpill.log)  |
| ceSteps | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1935/artifact/ceSteps.log)  |
| ceDigi | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1935/artifact/ceDigi.log)  |
| muDauSteps | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1935/artifact/muDauSteps.log)  |
| ceMix | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1935/artifact/ceMix.log)  |
| rootOverlaps | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1935/artifact/rootOverlaps.log)  |
| g4surfaceCheck | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1935/artifact/g4surfaceCheck.log)  |
| FIXME, TODO | :white_check_mark: | [TODO (0) FIXME (0) in 2 files](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1935/artifact/fixme_todo.log) |
| clang-tidy | :large_orange_diamond: | [0 errors 211 warnings](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1935/artifact/clang-tidy.log) |
| whitespace check | :white_check_mark: | [no whitespace errors found](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1935/artifact/whitespace_errs.log) |


N.B. These results were obtained from a build of this Pull Request at 949ee9a1dabde5114ad1083e35b7c2108ad0c77a after being merged into the base branch at 8926a71d94df388856f3bad5129def074131e278.


For more information, please check the job page [here](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1935/console).
Build artifacts are deleted after 5 days. If this is not desired, select `Keep this build forever` on the job page.

kutschke commented:
@brownd1978 A gentle reminder that I asked you to review this PR.  Please let me know when you expect to be able to get to it.  Thanks.

kutschke commented on RecoDataProducts/src/classes_def.xml:
```diff
@@ -245,6 +245,10 @@
 
+ <class name="mu2e::MVAResult"/>
+ <class name="mu2e::MVAResultCollection"/>
+ <class name="art::Wrapper<mu2e::MVAResultCollection>"/>
```
Do you expect to put a single MVAResult into the event as a data product, or only collections?  If you want the former, you also need the Wrapper<mu2e::MVAResult>.

AndrewEdmonds11 replied:
Done in 2abe4fb

kutschke replied:
Thanks!  One last CI run and I will merge.

kutschke commented:
@FNALbuild run build test

FNALbuild commented:
:hourglass: The following tests have been triggered for 2abe4fbb0f193b064e5368ef3b4666e1b04e1a1a: build  (Build queue is empty)

FNALbuild commented:
:sunny: The build tests passed at 2abe4fbb0f193b064e5368ef3b4666e1b04e1a1a.


| Test          | Result        | Details |
| ------------- |:-------------:| ------- |
| test with | :white_check_mark: | Command did not list any other PRs to include |
| merge | :white_check_mark: | Merged 2abe4fbb0f193b064e5368ef3b4666e1b04e1a1a at 8926a71d94df388856f3bad5129def074131e278 |
| build (prof) | :white_check_mark: | [Log file](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1940/artifact/scons.log). Build time: 11 min 04 sec |
| ceSimReco | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1940/artifact/ceSimReco.log)  |
| g4test_03MT | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1940/artifact/g4test_03MT.log)  |
| transportOnly | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1940/artifact/transportOnly.log)  |
| POT | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1940/artifact/POT.log)  |
| g4study | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1940/artifact/g4study.log)  |
| cosmicSimReco | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1940/artifact/cosmicSimReco.log)  |
| cosmicOffSpill | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1940/artifact/cosmicOffSpill.log)  |
| ceSteps | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1940/artifact/ceSteps.log)  |
| ceDigi | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1940/artifact/ceDigi.log)  |
| muDauSteps | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1940/artifact/muDauSteps.log)  |
| ceMix | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1940/artifact/ceMix.log)  |
| rootOverlaps | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1940/artifact/rootOverlaps.log)  |
| g4surfaceCheck | :white_check_mark: | [Log file.](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1940/artifact/g4surfaceCheck.log)  |
| FIXME, TODO | :white_check_mark: | [TODO (0) FIXME (0) in 2 files](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1940/artifact/fixme_todo.log) |
| clang-tidy | :large_orange_diamond: | [0 errors 211 warnings](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1940/artifact/clang-tidy.log) |
| whitespace check | :white_check_mark: | [no whitespace errors found](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1940/artifact/whitespace_errs.log) |


N.B. These results were obtained from a build of this Pull Request at 2abe4fbb0f193b064e5368ef3b4666e1b04e1a1a after being merged into the base branch at 8926a71d94df388856f3bad5129def074131e278.


For more information, please check the job page [here](https://buildmaster.fnal.gov/buildmaster/job/GitHubPRTests/job/mu2e-offline-build-test//1940/console).
Build artifacts are deleted after 5 days. If this is not desired, select `Keep this build forever` on the job page.
